src/plugins/catalog/store.go

```diff
@@ -41,3 +45,21 @@ func (catalogApi *CatalogApi) Get(catalogId uint64) (Catalog, error) {
 func (catalogApi *CatalogApi) Delete(catalogId uint64) error {
 	return catalogApi.DbClient.Delete(&Catalog{ID: catalogId}).Error
 }
+
+func (catalogApi *CatalogApi) Update(catalog *Catalog) error {
+	count := 0
+	if err := catalogApi.DbClient.
+		Model(&Catalog{}).
+		Where("name = ? AND id != ? AND user_id = ?", catalog.Name, catalog.ID, catalog.UserId).
```

Comment:
id != ? AND user_id = ? AND name = ?  might give MySQL hint for better performance.